pkg/cdi/cache.go

```diff
@@ -579,6 +579,14 @@ func (w *watch) update(dirErrors map[string]error, removed ...string) bool {
 		update bool
 	)
 
+	// If we failed to create an fsnotify.Watcher we have a nil watcher here
+	// (but with autoRefresh left on). One known case when this can happen is
+	// if we have too many open files. In that case we always return true and
+	// force a refresh.
+	if w.watcher == nil {
```

Comment:
Is ti possible to check it earlier, e.g. when `w.watcher` [is about to be assigned](https://github.com/cncf-tags/container-device-interface/blob/main/pkg/cdi/cache.go#L501) to `nil`? Or we want to keep it nil in a hope that it will be changed at some point?

Reply:
If I recall correctly we guard setting the watchers in a Mutex. Does that help to ensure this does not happen?

Reply:
> Is it possible to check it earlier, e.g. when `w.watcher` [is about to be assigned](https://github.com/cncf-tags/container-device-interface/blob/main/pkg/cdi/cache.go#L501) to `nil`? Or we want to keep it nil in a hope that it will be changed at some point?

Yes, I wanted to leave the door open for a better than the current recovery once we can again create fds (and that would probably be retrying a setup here).

In this very PR, I wanted to roll a minimum-footprint fix that we can review and merge quickly, and be fairly confident that it is safe to cherry pick and tag a v0.8.1.

Reply:
> If I recall correctly we guard setting the watchers in a Mutex. Does that help to ensure this does not happen?

This happens if we hit EMFILE while trying to create the watch.